### PR TITLE
[cmake] allow custom values for package name and package version

### PR DESCRIPTION
**Pull request by suveshpratapa**
@jwhui Let us know if you're okay with this small change.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -94,10 +94,13 @@ execute_process(
 )
```
Comment (bukepo) [inline CMakeLists.txt:94]: It seems not necessary to compute `OT_VERSION` when `OT_PACKAGE_VERSION` is defined? Should we remove guard this expression?

File: etc/cmake/options.cmake [hunk 2]
```diff
@@ -317,6 +317,16 @@ if(OT_UDP_FORWARD)
     target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_UDP_FORWARD_ENABLE=1")
 endif()
 
+option(OT_PACKAGE_NAME "enable vendor package name")
```
Comment (wgtdkp) [inline etc/cmake/options.cmake:320]: Should we use `variable` but not `option`? `option` is designed for boolean switches https://cmake.org/cmake/help/v3.10/command/option.html#option.

And better to have `set(OT_PACKAGE_NAME ...)` at beginning of `CMakeListstxt` but not this `options.cmake` file.

File: CMakeLists.txt (hunk 1 above)
Comment (suveshpratapa) [reply CMakeLists.txt]: Fixed! Thanks.

File: etc/cmake/options.cmake (hunk 2 above)
Comment (suveshpratapa) [reply etc/cmake/options.cmake]: Should be fixed now, thanks.

**Review by wgtdkp — COMMENTED** [review]
LGTM :+1: 
Thanks!

**Review by wgtdkp — APPROVED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -87,17 +87,19 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "(Apple)?[Cc]lang")
...
+set(OT_PACKAGE_VERSION "" CACHE STRING "OpenThread Package Version")
+if(OT_PACKAGE_VERSION STREQUAL "")
+    execute_process(
+        COMMAND bash "-c" "third_party/nlbuild-autotools/repo/scripts/mkversion -b ${OT_DEFAULT_VERSION}"
+        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        OUTPUT_VARIABLE OT_PACKAGE_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    message(STATUS "Setting default package version: ${OT_PACKAGE_VERSION}")
+endif()
+target_compile_definitions(ot-config INTERFACE "PACKAGE_VERSION=\"${OT_PACKAGE_VERSION}\"")
```
Comment (bukepo) [inline CMakeLists.txt:102]: print the version?
```
message(STATUS "Version: ${OT_PACKAGE_VERSION}")
```

**Review by bukepo — APPROVED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (suveshpratapa) [reply CMakeLists.txt:103]: Fixed now, thanks.

**Review by jwhui — APPROVED** [review]

**Pull request merged**